File: taches_yaml/generate_report.py
#!/usr/bin/env python3

import yaml
import os
import json

def read_yaml_file(file_path):
    """Read and parse a YAML file."""
    try:
        with open(file_path, 'r', encoding='utf-8') as file:
            return yaml.safe_load(file)
    except Exception as e:
        print(f"Error reading {file_path}: {e}")
        return None

def generate_progress_report():
    """Generate a progress report from all YAML files."""
    yaml_dir = os.path.dirname(os.path.abspath(__file__))
    yaml_files = [f for f in os.listdir(yaml_dir) if f.endswith('.yaml') and f != 'summary.yaml']
    
    total_tasks = 0
    completed_tasks = 0
    categories = []
    
    for file in yaml_files:
        file_path = os.path.join(yaml_dir, file)
        data = read_yaml_file(file_path)
        
        if data and 'tasks' in data:
            category_info = {
                'name': data.get('category', file.replace('.yaml', '')),
                'total': 0,
                'completed': 0,
                'subcategories': []
            }
            
            for subcategory, tasks in data['tasks'].items():
                sub_total = len(tasks)
                sub_completed = sum(1 for task in tasks if task.get('status') == 'completed')
                
                category_info['total'] += sub_total
                category_info['completed'] += sub_completed
                category_info['subcategories'].append({
                    'name': subcategory,
                    'total': sub_total,
                    'completed': sub_completed
                })
                
                total_tasks += sub_total
                completed_tasks += sub_completed
            
            categories.append(category_info)
    
    # Print summary
    print("=== Rapport de progression des tâches AgenticForge ===")
    print(f"Total des tâches: {total_tasks}")
    print(f"Tâches complétées: {completed_tasks}")
    print(f"Pourcentage: {completed_tasks/total_tasks*100:.1f}%\n")
    
    print("Détail par catégorie:")
    for category in categories:
        print(f"\n{category['name']}: {category['completed']}/{category['total']} "
              f"({category['completed']/category['total']*100:.1f}%)")
        for subcat in category['subcategories']:
            if subcat['total'] > 0:
                print(f"  {subcat['name']}: {subcat['completed']}/{subcat['total']} "
                      f"({subcat['completed']/subcat['total']*100:.1f}%)")
    
    # Save to JSON file
    report_data = {
        'summary': {
            'total_tasks': total_tasks,
            'completed_tasks': completed_tasks,
            'completion_percentage': round(completed_tasks/total_tasks*100, 2)
        },
        'categories': categories
    }
    
    with open(os.path.join(yaml_dir, 'progress_report.json'), 'w', encoding='utf-8') as f:
        json.dump(report_data, f, indent=2, ensure_ascii=False)
    
    print("\nRapport détaillé sauvegardé dans: progress_report.json")

if __name__ == "__main__":
    generate_progress_report()

File: taches_yaml/generate_report.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const yaml = require('js-yaml');

// Function to read and parse YAML file
function readYamlFile(filePath) {
  try {
    const fileContents = fs.readFileSync(filePath, 'utf8');
    return yaml.load(fileContents);
  } catch (error) {
    console.error(`Error reading ${filePath}:`, error);
    return null;
  }
}

// Function to generate a simple HTML report
function generateHtmlReport(categories) {
  let html = `
<!DOCTYPE html>
<html>
<head>
    <title>Rapport des tâches AgenticForge</title>
    <style>
        body { font-family: Arial, sans-serif; margin: 20px; }
        h1 { color: #333; }
        h2 { color: #666; border-bottom: 1px solid #eee; }
        .category { margin-bottom: 30px; }
        .task { margin: 5px 0; padding: 5px; background: #f5f5f5; }
        .completed { background: #d4edda; }
        .pending { background: #f8d7da; }
        .stats { font-weight: bold; margin: 10px 0; }
    </style>
</head>
<body>
    <h1>Rapport des tâches AgenticForge</h1>
`;

  categories.forEach((category) => {
    html += `
    <div class="category">
        <h2>${category.category || category.name}</h2>
        <div class="stats">
            Total: ${category.tasks ? Object.values(category.tasks).flat().length : 'N/A'} | 
            Complétées: ${
              category.tasks
                ? Object.values(category.tasks)
                    .flat()
                    .filter((t) => t.status === 'completed').length
                : 'N/A'
            }
        </div>
`;

    if (category.tasks) {
      Object.entries(category.tasks).forEach(([subcategory, tasks]) => {
        html += `<h3>${subcategory}</h3>`;
        tasks.forEach((task) => {
          html += `<div class="task ${task.status}">[${task.id}] ${task.description} (${task.status})</div>`;
        });
      });
    }

    html += `</div>`;
  });

  html += `
</body>
</html>`;

  return html;
}

// Main function
function main() {
  const yamlDir = path.join(__dirname);
  const yamlFiles = fs
    .readdirSync(yamlDir)
    .filter((file) => file.endsWith('.yaml') && file !== 'summary.yaml');

  const categories = [];

  yamlFiles.forEach((file) => {
    const filePath = path.join(yamlDir, file);
    const data = readYamlFile(filePath);
    if (data) {
      categories.push(data);
    }
  });

  // Generate HTML report
  const htmlReport = generateHtmlReport(categories);
  fs.writeFileSync(path.join(__dirname, 'report.html'), htmlReport);
  console.log('Rapport HTML généré: report.html');

  // Generate simple console report
  console.log('\n=== Rapport des tâches ===');
  categories.forEach((category) => {
    if (category.tasks) {
      const totalTasks = Object.values(category.tasks).flat().length;
      const completedTasks = Object.values(category.tasks)
        .flat()
        .filter((t) => t.status === 'completed').length;
      console.log(
        `${category.category}: ${completedTasks}/${totalTasks} tâches complétées`,
      );
    }
  });
}

main();
